### PR TITLE
Feature/upgrade logging collections history details

### PR DESCRIPTION
**Pull request by mikeAdamss**
### What

More api logging upgrades.

### How to review

Sanity check, try it.

### Who can review

Anyone.

### CODE REVIEW

File: zebedee-cms/src/main/java/com/github/onsdigital/zebedee/api/CollectionDetails.java [hunk 1]
```diff
@@ -112,9 +112,13 @@ public CollectionDetail get(HttpServletRequest request, HttpServletResponse resp
         result.teamsDetails = zebedeeCmsService.getZebedee().getTeamsService().resolveTeamDetails(teamIds);
         result.teamsDetails.forEach(team -> collection.getDescription().getTeams().add(team.getName()));
 
+        String collectionId = Collections.getCollectionId(request);
+
         if (datasetImportEnabled) {
-            logInfo("CollectionDetails GET endpoint: datasetImportEnabled including dataset and dataset version " +
-                    "details to response").collectionId(collection).user(session).log();
+            info().data("collectionId", collectionId).data("user", session)
```
Comment (daiLlew) [inline zebedee-cms/src/main/java/com/github/onsdigital/zebedee/api/CollectionDetails.java]: `session.getEmail()`

File: zebedee-cms/src/main/java/com/github/onsdigital/zebedee/api/CollectionHistory.java [hunk 2]
```diff
@@ -60,9 +60,7 @@ private void checkPermission(Session session) throws ZebedeeException {
                 throw new UnauthorizedException("You are not authorised to create collections.");
             }
         } catch (IOException io) {
-            logError(io, "Unexpected error while trying to access permissions")
-                    .user(session.getEmail())
-                    .logAndThrow(UnexpectedErrorException.class);
+            error().data("user", session).logAndThrow (io, "Unexpected error while trying to access permissions");
```
Comment (daiLlew) [inline zebedee-cms/src/main/java/com/github/onsdigital/zebedee/api/CollectionHistory.java]: `session.getEmail()`

Comment (daiLlew) [reply zebedee-cms/src/main/java/com/github/onsdigital/zebedee/api/CollectionHistory.java]: I think this needs to have `throw` in front - the new log n throw method doesn't throw an exception is returns an exception so you have to declare a throw in order for it to happen... possibly a better name for this method is needed.

```
throw error().data("user", session).logAndThrow (io, "Unexpected error while trying to access permissions");
```

Comment (mikeAdamss) [reply zebedee-cms/src/main/java/com/github/onsdigital/zebedee/api/CollectionHistory.java]: probably ok as you had it, this is more a bad assumption on my part that a code issue.

**Pull request merged**